### ISSUE TITLE
Implement TOUCH command

### ISSUE THREAD

**Issue by kelvinmwinuka** [EchoVault/SugarDB#60]
Implement the TOUCH command, which updates the access times of the provided keys. 
Reference: https://redis.io/docs/latest/commands/touch/

## Client-Server Spec:

Command File: ./internal/modules/generic/commands.go
Test File: ./internal/modules/generic/commands_test.go

Command: `touch`
Module: `constants.GenericModule`
Categories: `constants.KeyspaceCategory`, `constants.ReadCategory`, `constants.FastCategory`
Description: `(TOUCH key [key ...]) Alters the last access time of a key(s). A key is ignored if it does not exist.`
Sync: `true`

## Embedded Spec:

Command File: ./echovault/api_generic.go
Test File: ./echovault/api_generic_test.go

NOTE: You may have to edit the `HandlerFuncParams` type to include a function that touches the keys.

**Comment by osteensco** [comment]
I'll pick this one up next

**Comment by osteensco** [comment]
@kelvinmwinuka I'm thinking this just needs to use/extend updateKeysInCache but not 100% certain on the ideal route to go. Did you have something else in mind for implementing this?

**Comment by kelvinmwinuka** [comment]
> @kelvinmwinuka I'm thinking this just needs to use/extend updateKeysInCache but not 100% certain on the ideal route to go. Did you have something else in mind for implementing this? 

I think that's a reasonable approach.

**Comment by osteensco** [comment]
I'm implementing OBJECT IDLETIME and OBJECT FREQ in addition to TOUCH since these are dependencies of TOUCH to enable better testing.

@kelvinmwinuka this may need a separate issue opened, but I ran into something with the `adjustMemoryUsage` EchoVault method while building out the tests for these commands. 
I set the max memory of the test server to 1mb assuming this should be the max size of the cache, but we are comparing to HeapInuse which is total memory of the program (I think), so this fails to return nil for low MaxMemory allocations (under 2mb). Is this intended?

```golang
	// If we're using less memory than the max-memory, there's no need to evict.
	if memStats.HeapInuse < server.config.MaxMemory {
		return nil
	}
```

along the same vein, we may want some sort of logic check config settings are passed because as it is now it's possible to pass in too low of a max memory usage as well as eviction interval for the server to properly function with any eviction policy.
